pkg/controller/storagecluster/reconcile.go

```diff
@@ -176,12 +201,23 @@ func (r *ReconcileStorageCluster) Reconcile(request reconcile.Request) (reconcil
 		// to set readiness.
 		r := ready.NewFileReady()
 		err = r.Set()
-		instance.Status.Phase = statusutil.PhaseReady
 		if err != nil {
 			reqLogger.Error(err, "Failed to mark operator ready")
 			return reconcile.Result{}, err
 		}
+		if conditionsv1.IsStatusConditionTrue(instance.Status.Conditions, conditionsv1.ConditionAvailable) &&
```

Comment:
Inside the if r.conditions== nil block, the cluster will always be in a happy state. Your 'else' block will never be hit and 'if' checks are redundant. What you had before should be fine.
`instance.Status.Phase = statusutil.PhaseReady`.

Reply:
ack.